src/guided_fuzzing_daemon/storage.py

```diff
@@ -266,22 +288,25 @@ def iter_projects(self, prefix: str = "") -> Iterable[str]:
                 yield obj["Prefix"][:-1]  # trim trailing delimiter
 
     def __getitem__(self, name: PurePosixPath) -> CloudStorageFile:
-        return S3File(name, None, self)
+        return S3File(name, None, None, self)
 
 
 class GCSFile(CloudStorageFile):
     def __init__(
         self,
         name: PurePosixPath,
         modified: datetime | None,
+        size: int | None,
         _provider: GoogleCloudStorage,
     ) -> None:
-        super().__init__(name, modified)
+        super().__init__(name, modified, size)
         self._provider = _provider
 
     def _refresh(self) -> None:
         blob = self._provider.bucket.blob(str(self.path))
+        blob.reload()
```

Comment:
Did it not fetch metadata before without `reload()`, or does this prevent two requests being made?

Reply:
When I originally wrote this, I believed that calling reload was necessary to ensure that the metadata was populated.  However, looking back at it now, I cannot see a situation where this will occur.  For the time being, I've removed this call in 7c1881fd4f45efc40d12084e7034462f18922914.  We can add it back if necessary.